app/controllers/reset_passwords_controller.rb

```diff
@@ -0,0 +1,24 @@
+class ResetPasswordsController < ApplicationController
+  def show
+    user = User.find_by(token: params[:id])
+    if user
+      @token = params[:id]
+    else
+      redirect_to expired_token_path
+    end
+  end
+
+  def create
+    token = params[:token]
+    redirect_to root_path and return if token.blank?
+
+    user = User.find_by(token: params[:token])
+    if user
+      user.update!({password: params[:password], token: nil})

```

Comment:
If the password is incorrect, or invalid - how are you going to handle that situation? It should check at the `if` statement and you should create a branch for unhappy path.